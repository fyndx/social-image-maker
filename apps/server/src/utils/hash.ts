import { CryptoHasher } from "bun";

export const getSha256Hash = (input: string): string => {
  const textEncoder = new TextEncoder();
  const encodedInput = textEncoder.encode(input);
  const hasher = new CryptoHasher("sha256");
  hasher.update(encodedInput);
  return hasher.digest("hex");
};
