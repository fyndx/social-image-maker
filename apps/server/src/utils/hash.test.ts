import {describe, expect, test} from 'bun:test'
import { getSha256Hash } from './hash'

describe('getSha256Hash', () => {
  test('hash string', () => {
    const url = 'https://fyndx.io'
    expect(getSha256Hash(url)).toMatchSnapshot()
  })
})
