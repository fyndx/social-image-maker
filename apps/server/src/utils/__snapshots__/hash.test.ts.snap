// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`getSha256Hash hash string 1`] = `"373c78fbd0749c68178c505ee70d3094d05acf3e12562830cccc7c46b3f09a06"`;
